javalib/src/main/scala/java/util/StringJoiner.scala

```diff
@@ -0,0 +1,61 @@
+// Ported from Scala.js commit: 57d71da dated: 2023-05-31
+
+package java.util
+
+@inline
+final class StringJoiner private (
+    delimiter: String,
+    prefix: String,
+    suffix: String
+) extends AnyRef {
+
+  /** The custom value to return if empty, set by `setEmptyValue` (nullable).
+   *
+   *  If `null`, defaults to `prefix + suffix`.
+   */
+  private var emptyValue: String = null
+
+  /** The current value, excluding prefix and suffix. */
+  private var value: String = ""
+
+  /** Whether the string joiner is currently empty. */
+  private var isEmpty: Boolean = true
+
+  def this(delimiter: CharSequence) =
+    this(delimiter.toString(), "", "")
+
+  def this(
+      delimiter: CharSequence,
+      prefix: CharSequence,
+      suffix: CharSequence
+  ) =
+    this(delimiter.toString(), prefix.toString(), suffix.toString())
+
+  def setEmptyValue(emptyValue: CharSequence): StringJoiner = {
+    this.emptyValue = emptyValue.toString()
+    this
+  }
+
+  override def toString(): String =
+    if (isEmpty && emptyValue != null) emptyValue
+    else prefix + value + suffix
+
+  def add(newElement: CharSequence): StringJoiner = {
+    if (isEmpty)
+      isEmpty = false
+    else
+      value += delimiter
+    value += newElement // if newElement is null, adds "null"
```

Comment:
Scala.js has used string concat here becouse string are native construct in JS runtime, and this action does not contain performance penalty. However, on JVM or Native we don't have native String primitives and their concatontation is expensive. We should use `java.lang.StringBuilder` instead.

Reply:
Something like this?
```scala
    val builder = java.lang.StringBuilder(value)
    if (isEmpty)
      isEmpty = false
    else
      builder.append(delimiter)

    if (newElement == null)
      builder.append("null")
    else
      builder.append(newElement)

    value = builder.toString()
    this
```

Reply:
yeah, something like that. Should do the job. We probably don't need to store result in value